File: go/src/Core/Logger.go
package Core

import (
	"encoding/json"
	"os"
	"time"
)

var logFile = "storage/logs/app.log"

func Info(message string) {
	write("INFO", message, nil, nil)
}

func Error(message string) {
	write("ERROR", message, nil, nil)
}

func write(level string, message string, meta map[string]interface{}, data map[string]interface{}) {
	log := map[string]interface{}{
		"timestamp": time.Now().Format(time.RFC3339),
		"message":   message,
		"level":     level,
		"trace_id":  generateTraceID(),
		"service":   "users-api",
		"meta_data": meta,
		"data":      data,
	}

	f, _ := os.OpenFile(logFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
	defer f.Close()

	json.NewEncoder(f).Encode(log)
}

func generateTraceID() string {
	return randomHex(8) + "," + randomHex(8)
}

func randomHex(n int) string {
	bytes := make([]byte, n)
	for i := range bytes {
		bytes[i] = byte(65 + i%26)
	}
	return string(bytes)
}

File: php/src/Core/Logger.php
<?php

namespace Volley5544\PhpProject\Core;

class Logger
{
    protected static string $logFile = __DIR__ . '/../../storage/logs/app.log';

    public static function info(
        string $message,
        array $metaData = [],
        array $data = []
    ) {
        self::write("INFO", $message, $metaData, $data);
    }

    public static function error(
        string $message,
        array $metaData = [],
        array $data = []
    ) {
        self::write("ERROR", $message, $metaData, $data);
    }

    protected static function write(
        string $level,
        string $message,
        array $metaData = [],
        array $data = []
    ) {
        $log = [
            "timestamp" => date("Ymd\TH:i:sP"), // ISO8601 with timezone
            "message" => $message,
            "level" => $level,
            "trace_id" => self::generateTraceId(),
            "service" => "users-api",
            "meta_data" => (object)$metaData,
            "data" => (object)$data,
        ];

        file_put_contents(
            self::$logFile,
            json_encode($log, JSON_UNESCAPED_UNICODE) . PHP_EOL,
            FILE_APPEND
        );
    }

    protected static function generateTraceId(): string
    {
        return bin2hex(random_bytes(8)) . ',' . bin2hex(random_bytes(8));
    }
}